lib/zendesk_api/middleware/response/parse_iso_dates.rb

```diff
@@ -19,7 +19,9 @@ def parse_dates!(value)
           case value
           when Hash then value.each { |key,element| value[key] = parse_dates!(element) }
           when Array then value.each_with_index { |element, index| value[index] = parse_dates!(element) }
-          when /\A\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}Z\Z/m then Time.parse(value)
+          when String
+            return Time.parse(value) if value.valid_encoding? && value =~ /\A\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}Z\Z/m
+            value
```

Comment:
If no one really sees any possible case that this data (bad characters) might be of any interest to a client, lets just sanitise the data.
@steved when we fixed this on server side was it due to bad characters sent from a client, or just handling the data bad on server side?
Is there any chance this can be an emoji?